sycl/include/CL/sycl/accessor.hpp

```diff
@@ -682,9 +687,14 @@ class accessor :
   using AccessorSubscript =
       typename AccessorCommonT::template AccessorSubscript<Dims>;
 
-  using RefType = DataT &;
   using ConcreteASPtrType = typename detail::PtrValueType<DataT, AS>::type *;
+#if defined(RESTRICT_WRITE_ACCESS_TO_CONSTANT_PTR)
```

Comment:
I think we use these definitions even if `RESTRICT_WRITE_ACCESS_TO_CONSTANT_PTR` is not defined.
```c++
  using RefType = detail::const_if_const_AS<AS, DataT> &;
  using PtrType = detail::const_if_const_AS<AS, DataT> *;
```

Thanks to definition:
```c++
#if defined(RESTRICT_WRITE_ACCESS_TO_CONSTANT_PTR)
template <access::address_space AS, class DataT>
using const_if_const_AS =
    typename std::conditional<AS == access::address_space::constant_space,
                              const DataT, DataT>::type;
#else
template <access::address_space AS, class DataT>
using const_if_const_AS = DataT;
#endif
```
Right?
If so, I think there are a few other places where it applies as well.

Reply:
Sure, done.